unity/Assets/Maroon/reusablePrefabs/NewRuler/Scripts/Test_RulerDisplay.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Assets.Maroon.reusablePrefabs.NewRuler.Scripts;
+using Maroon.GlobalEntities;
+using Maroon.Physics.CoordinateSystem;
+using TMPro;
+using UnityEngine;
+using Maroon.GlobalEntities;
+
+public class Test_RulerDisplay : MonoBehaviour
+{
+    [SerializeField] private RulerLogic ruler;
+
+    [SerializeField] private TMP_Text DistanceText;
+
+    [SerializeField] private GameObject StartPositionX;
+    [SerializeField] private GameObject StartPositionY;
+    [SerializeField] private GameObject StartPositionZ;
+
+    [SerializeField] private GameObject EndPositionX;
+    [SerializeField] private GameObject EndPositionY;
+    [SerializeField] private GameObject EndPositionZ;
+
+    [SerializeField] private GameObject StartPositionXUnit;
+    [SerializeField] private GameObject StartPositionYUnit;
+    [SerializeField] private GameObject StartPositionZUnit;
+
+    [SerializeField] private GameObject EndPositionXUnit;
+    [SerializeField] private GameObject EndPositionYUnit;
+    [SerializeField] private GameObject EndPositionZUnit;
+    
+    private Vector3 startMeasuringPosition;
+    private Vector3 endMeasuringPosition;
+
+    public Vector3 StartMeasuringPosition => CoordSystemHelper.Instance.GetSystemPosition(ruler.RulerStart.transform.position);
+
+    public Vector3 EndMeasuringPosition => CoordSystemHelper.Instance.GetSystemPosition(ruler.RulerEnd.transform.position);
+
+    //TODO: Third parameter of CheckVariable should be the select script e.g. ruler.RulerStart.GetComponent<PC_SelectScript>() to invoke coloumbs law change logic.
+    private void Start()
+    {
+        StartPositionX.GetComponent<PC_InputParser_Float_TMP>().onValueChangedFloat.AddListener(endVal =>
```

Comment:
Do we really need this InputParser stuff?
Just use the OnEndEdit Event from the InputField type

Reply:
Currently used to limit the users input e.g. 0-1000. The InputFieldEditor could be extended by this functionality